core/comparison.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2019 Scipp contributors (https://github.com/scipp)
+/// @file
+
+#include "scipp/core/comparison.h"
+
+namespace scipp::core {
+
+/**
+ * Tests if the unit, values (and variances where appropriate) of two Variables
+ * are within an absolute tolerance.
+ */
+bool is_approx(Variable a, const Variable &b, const double tol) {
+  bool match{true};
```

Comment:
Should return `false` if `dtype` does not match, otherwise you get weird errors instead of returning?